### DISCUSSION TITLE
appium.io changing my capabilities for WinAppDriver and therefore not working

### DISCUSSION THREAD

**Discussion by bistritapcv** [appium/appium#18503] [Q&A]
I am using (trying to use) WinAppDriver with Java and Eclipse and Selenium on Windows 10 for just one part of Selenium/IE-Edge where Selenium cannot handle xpaths (for instance a Windows login box).  I did try https://user:password@url.com for the URL to no avail.  I got this to work in Winium but they will not let us use Winium as there has not been an update lately.  When the login box pops up I need to use WinAppDriver.

All the documentation says you need to run appium.io to communicate with WinAppDriver.  So I followed all the documentation, started WinAppDriver and then did

      DesiredCapabilities dcaps = new DesiredCapabilities();
      dcaps.setCapability("automationName", "Windows");
      dcaps.setCapability("appId", "{1AC14E77-02E7-4E5D-B744-2EB1AE5198B7}\\notepad.exe}");
      dcaps.setCapability("app", "C:\\Windows\\System32\\notepad.exe");
      dcaps.setCapability("platformName", "Windows");
     dcaps.setCapability("deviceName", "Windows PC");
     WindowsDriver app = new WindowsDriver(new URL("http://127.0.0.1:4723/"), dcaps);

I get "Could not start a new session. Response code 400. Message: Missing command parameter: appId ".
I know exactly what the problem is.  On the WinAppDriver window I see the capabilities being passed are "appium:app, appium:appId" etc.  That is, all the capabilities have "appium:" prefixed to them.

I stepped through the code and even found the place where it does it.  There is a method called 
W3CSafe(capabilities) which gets called from startSession(), which changes the capability names.  It is as if it is expecting WinAppDriver to expect "appium:" prefixed to the capability names, but WinAppDriver is expecting without the prefixes.

I have tried several different versions of appium.io.  As for WinAppDriver, corporate installs that so I do not have control but it appears to be correct.

Anyway, I want to figure out how to resolve this?  Maybe there is an appium.io version which does not prefix the capabilities?  Maybe there are options when you start WinAppDriver though i cannot find any.  I tried WinAppDriver --help, /help, ? and "man WinAppDriver" which I did not expect to work on Windows, and it didn't.

Can anyone offer any suggestions?  I really wanted to add this as an issue but I could not find the right place.

**Comment by jlipps** [comment]
Are you running WAD yourself or via the appium-windows-driver

**Reply by bistritapcv** [reply]
I tried something before that I did not think worked, but I tried it again and it seems to.
Instead of "WindowsDriver app = new WindowsDriver(exec, dcaps);"

     WindowsDriver app = new WindowsDriver(new URL("http://127.0.0.1:4723/"), dcaps);

I do a

     HttpCommandExecutor exec = new HttpCommandExecutor(new URL("http://127.0.0.1:4723/"));
     WindowsDriver app = new WindowsDriver(exec, dcaps);

I did not think it worked because I tried app.findElement(By.xpath("//*[@name = 'Text Editor']")) which gave unimplemented method error.  So I thought I had to write all my methods, but in the appium driver you calso cannot find by xpath.  Just by name and probably others (I haven't tried).

**Reply by jlipps** [reply]
Great glad to hear it's working!

**Reply by bistritapcv** [reply]
How important is developer mode?

**Comment by bistritapcv** [comment]
I double click on wad from file explorer so it runs and then in selenium do appium.io driver with the url. I forget the port. Something like 4723.
Is there another way?

Sent from Yahoo Mail on Android 
 
  On Thu, Apr 13, 2023 at 19:10, Jonathan ***@***.***> wrote:   


Are you running WAD yourself or via the appium-windows-driver
